envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadataTest.kt

```diff
@@ -226,6 +259,167 @@ class NodeMetadataTest {
+        assertThat(outgoing.serviceDependencies).isEmpty()
+    }
+
+    @Test
+    fun `should parse service dependencies and for missing config use config defined in allServiceDependency`() {
+        // given
+        val proto = outgoingDependenciesProto {
+            withService(serviceName = "*", idleTimeout = "10s", requestTimeout = "10s")
+            withService(serviceName = "service-name-1", idleTimeout = "5s", requestTimeout = null)
+            withService(serviceName = "service-name-2", idleTimeout = null, requestTimeout = "4s")
+            withService(serviceName = "service-name-3", idleTimeout = null, requestTimeout = null)
+        }
+        val properties = snapshotProperties(allServicesDependenciesIdentifier = "*")
+        // when
+
+        val outgoing = proto.toOutgoing(properties)
+
+        // expects
+        assertThat(outgoing.allServicesDependencies).isTrue()
+        assertThat(outgoing.defaultServiceSettings).hasTimeouts(idleTimeout = "10s", requestTimeout = "10s")
+
+        outgoing.serviceDependencies.assertServiceDependency("service-name-1")
+            .hasTimeouts(idleTimeout = "5s", requestTimeout = "10s")
+        outgoing.serviceDependencies.assertServiceDependency("service-name-2")
+            .hasTimeouts(idleTimeout = "10s", requestTimeout = "4s")
+        outgoing.serviceDependencies.assertServiceDependency("service-name-3")
+            .hasTimeouts(idleTimeout = "10s", requestTimeout = "10s")
+    }
+
+    @Test
+    fun `should parse service dependencies and for missing configs use config defined in properties when allServiceDependency isn't defined`() {
+        // given
+        val proto = outgoingDependenciesProto {
+            withService(serviceName = "service-name-1", idleTimeout = "5s", requestTimeout = null)
+            withService(serviceName = "service-name-2", idleTimeout = null, requestTimeout = "4s")
+            withService(serviceName = "service-name-3", idleTimeout = null, requestTimeout = null)
+        }
+        val properties = snapshotProperties(idleTimeout = "12s", requestTimeout = "12s")
+        // when
+
+        val outgoing = proto.toOutgoing(properties)
+
+        // expects
+        assertThat(outgoing.allServicesDependencies).isFalse()
+        assertThat(outgoing.defaultServiceSettings).hasTimeouts(idleTimeout = "12s", requestTimeout = "12s")
+
+        outgoing.serviceDependencies.assertServiceDependency("service-name-1")
+            .hasTimeouts(idleTimeout = "5s", requestTimeout = "12s")
+        outgoing.serviceDependencies.assertServiceDependency("service-name-2")
+            .hasTimeouts(idleTimeout = "12s", requestTimeout = "4s")
+        outgoing.serviceDependencies.assertServiceDependency("service-name-3")
+            .hasTimeouts(idleTimeout = "12s", requestTimeout = "12s")
+    }
+
+    @Test
+    fun `should parse domain dependencies and for missing config use config defined in properties even if allServiceDependency is defined`() {
```

Comment:
This is a case we did not see in the design docs :| isn't this going to be confusing for the user?

Reply:
This is not yet put in design docs because I have found this use case recently.

```
metadata:
  proxy_settings:
    outgoing:
      dependencies:
        - service: "*"
          timeoutPolicy:
            idleTimeout: 400
            requestTimeout: 800
        - domain: "http://some-domain"
          timeoutPolicy:
            idleTimeout: 400
```

It seems to me that value of requestTimeout for "http://some-domain" should be taken from properties because "*" specified all services and not for domains.

Reply:
Ok :) I think people will try to do:

```
        - domain: "*"
          timeoutPolicy:
            idleTimeout: 401
            requestTimeout: 802
```

To set a default, we should make sure there is a clear error message that this only works for service dependencies.

Reply:
I added code which throw NodeMetadataValidationException when sb use * for domain